openmc/stats/univariate.py

```diff
@@ -0,0 +1,344 @@
+from abc import ABCMeta, abstractmethod
+from collections import Iterable
+from numbers import Real
+import sys
+from xml.etree import ElementTree as ET
+
+import openmc.checkvalue as cv
+
+if sys.version_info[0] >= 3:
+    basestring = str
+
+
+class Univariate(object):
+    """Probability distribution of a single random variable.
+
+    The Univariate class is an abstract class that can be derived to implement a
+    specific probability distribution.
+
+    Parameters
+    ----------
+    name : str
+        Name of the distribution
+
+    Attributes
+    ----------
+    name : str
+        Name of the distributions
+
+    """
+
+    __metaclass__ = ABCMeta
+
+    def __init__(self, name=None):
+        self._name = None
+        if name is not None:
+            self.name = name
+
+    @property
+    def name(self):
+        return self._name
+
+    @name.setter
+    def name(self, name):
+        cv.check_type('name', name, basestring)
+        self._name = name
+
+    @abstractmethod
+    def to_xml(self):
+        return ''
+
+
+class Discrete(Univariate):
+    """Distribution characterized by a probability mass function.
```

Comment:
If so, I think a class name `Multinomial` would be more informative and transparent than `Discrete`